timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -952,3 +877,23 @@ def convert_frequency(
         )
         resampled_df.static_features = self.static_features
         return resampled_df
+
+    def __dir__(self) -> List[str]:
+        # This hides method from IPython autocomplete, but not VSCode autocomplete
+        deprecated = ["get_reindexed_view", "to_regular_index"]
+        return [d for d in super().__dir__() if d not in deprecated]
```

Comment:
@Innixma Here is a partial solution that I found for hiding some methods from autocomplete (inspired by [this SO answer](https://stackoverflow.com/questions/2531479/hide-deprecated-methods-from-tab-completion)). I verified that this works for IPython, but it doesn't work for VSCode/Intellisense, unfortunately.

Please let me know if you find anything more general that deals with all autocomplete engines.

Reply:
nice! but why is it `super().__dir__` and not `self.__dir__` ? also, is there a solution to hardcoding the method names?

Reply:
I think `self.__dir__` would just go into infinite recursion, did you mean `self.__dict__.keys()` or something like that? `super().__dir__` should take the default implementation of the dir method by `object`, as it's the only parent class that implements it.

Could you elaborate on hardcoding the method names?